assets/js/googlesitekit/api/index.js

```diff
@@ -1,5 +1,79 @@
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { deleteItem, getItem, getKeys, setItem } from './cache';
+import { createCacheKey } from './index.private';
+
+// Caching is enabled by default.
 let cachingEnabled = true;
 
+/**
+ *
+ * @param {string} type        The data to access. One of 'core' or 'modules'.
+ * @param {string} identifier  The data identifier, eg. a module slug like `'search-console'`.
+ * @param {string} datapoint   The endpoint to request data from.
+ * @param {Object} queryParams Query params to send with the request.
+ * @param {Object} data        Request data to send.
+ */
+const siteKitRequest = async ( {
+	cacheTTL = 3600,
+	data,
+	datapoint,
+	identifier,
+	method = 'GET',
+	queryParams,
+	type,
+	useCache = undefined,
+} = {} ) => {
+	invariant( type, '`type` argument for requests is required.' );
+	invariant( identifier, '`identifier` argument for requests is required.' );
+	invariant( datapoint, '`datapoint` argument for requests is required.' );
+
+	const useCacheForRequest = useCache !== undefined ? useCache : usingCache();
+	const cacheKey = createCacheKey( type, identifier, datapoint, queryParams );
+
+	if ( useCacheForRequest ) {
+		const { cacheHit, value } = await getItem( cacheKey, cacheTTL );
+
+		if ( cacheHit ) {
+			return value;
+		}
+	}
+
+	// Make an API request to retrieve the results.
+	try {
+		const response = await apiFetch( {
+			data,
+			method,
+			path: addQueryArgs(
+				`/google-site-kit/v1/${ type }/${ identifier }/data/${ datapoint }`,
+				queryParams
+			),
+		} );
+
+		if ( useCacheForRequest ) {
+			await setItem( cacheKey, response );
+		}
+
+		return response;
+	} catch ( error ) {
+		// global.console.error( 'Google Site Kit API Error', error );
```

Comment:
We probably need to expand this going forward, but until we start _actually_ using this code in the project I didn't want to expand on the error handling too much. I think it'd be likely that we'd end up rewriting/tweaking it and I'd rather that part come about organically.